src/EditorFeatures/Core/Implementation/SmartIndent/AbstractIndentationService.cs

```diff
@@ -30,13 +31,16 @@ private IEnumerable<IFormattingRule> GetFormattingRules(Document document, int p
         public IndentationResult? GetDesiredIndentation(
             Document document, int lineNumber, CancellationToken cancellationToken)
         {
-            var root = document.GetSyntaxRootSynchronously(cancellationToken);
-            var sourceText = root.SyntaxTree.GetText(cancellationToken);
-            var documentOptions = document.GetOptionsAsync(cancellationToken).WaitAndGetResult(cancellationToken);
-
-            var lineToBeIndented = sourceText.Lines[lineNumber];
+            GetIndenterData(document, lineNumber,
```

Comment:
💭 I'm not a big fan of calling `GetIndenterData` twice on this path (which seems like it would be a common entry point for this interface), but I also haven't measured the impact specifically.

Reply:
Have fixed things up to only get the indenter once.